### PR TITLE
Jirachi and Meta Groudon Quest

### PR DESCRIPTION
**Pull request by Farboo171**
A Quest for Jirachi and Meta Groudon

This quest is balanced to be post-E4 content for Hoenn. The first quest is to obtain Jirachi, and follows the plot of the movie Jirachi: Wish Maker.

The second quest is to provide an obtain method for Meta Groudon as seen in the movie and previous quest. This quest requires completing the Jirachi quest, and having a Groudon.

TODO:
1. Crew Testing
2. Clown Team Rocket image

### CODE REVIEW

File: src/scripts/temporaryBattle/TemporaryBattleList.ts [hunk 1]
```diff
@@ -762,6 +762,66 @@ TemporaryBattleList['Wally 2'] = new TemporaryBattle(
...
+    undefined,
+    {
+        displayName: 'Butler',
+        imageName: '../npcs/Butler',
+    }
+);
+TemporaryBattleList['Meta Groudon'] = new TemporaryBattle(
+    'Meta Groudon',
+    [new GymPokemon('Meta Groudon', 13800000, 100)],
+    '<i>Meta Groudon lets out a gutteral roar and melts back into the earth!</i>',
```
Comment (Vodovik) [inline src/scripts/temporaryBattle/TemporaryBattleList.ts:815]: guttural*

File: src/scripts/quests/QuestLineHelper.ts [hunk 2]
```diff
@@ -721,6 +721,83 @@ class QuestLineHelper {
...
+        const talkToButler1 = new TalkToNPCQuest(Butler1, 'Learn the legend of the Millennium Comet from Butler near Lavaridge Town.');
+        jirachiQuestLine.addQuest(talkToButler1);
+
+        const clearMtChimney2 = new CustomQuest(1, 0, 'Climb to the Mt. Chimney Crater to get a better view of the Millennium Comet as it passes.', () => App.game.statistics.dungeonsCleared[GameConstants.getDungeonIndex('Mt. Chimney Crater')]());
+        jirachiQuestLine.addQuest(clearMtChimney2);
+
+        const catchAbsol = new CaptureSpecificPokemonQuest('Absol', 'You are being stalked by Absol, the Disaster Pokémon. Capture it.', 1, true);
+        jirachiQuestLine.addQuest(catchAbsol);
+
+        const coccoonHatch = new TalkToNPCQuest(CoccoonHatch, 'Examine the crystalline coccoon Butler gave you while at the Mt. Chimney Crater.');
```
Comment (Vodovik) [inline src/scripts/quests/QuestLineHelper.ts:744]: this way of writing cocoon comes from games/movie?

Comment (miaiam2006) [reply src/scripts/quests/QuestLineHelper.ts:744]: Officially it is in fact “Cocoon”, from checking the poster art on [bulbapedia](https://m.archives.bulbagarden.net/media/upload/e/ed/M17_English_poster.png)

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**